prime-router/src/main/kotlin/fhirengine/translation/hl7/ConfigSchemaProcessor.kt

```diff
@@ -73,10 +71,8 @@ abstract class ConfigSchemaProcessor : Logging {
         }
 
         // when valueSet is available, return mapped value or null if match isn't found
-        // does a lowerCase comparison
-        if (retVal != null && element.valueSet.isNotEmpty()) {
-            val lowerSet = element.valueSet.mapKeys { it.key.lowercase() }
-            val valStr = lowerSet[retVal?.primitiveValue()?.lowercase() ?: ""]
+        if (retVal != null && element.valueSet != null) {
+            val valStr = element.valueSet?.getMappedValue(retVal?.primitiveValue() ?: "")
```

Comment:
Same question here

Reply:
This one's a little more difficult to unpack, as the logic differs from the other use of `valueSet` above in that it returns `null` if `valueSet` is present but has no match, but if `valueSet` is not present than it returns `retVal`. The difference is down to the two methods having different consumers and use cases, though perhaps the behavior of both should be explored further.

Reply:
I think though this is the same situation though where the the null check on line 74 isn't required

Reply:
It's materially different because `getMappedValue` also returns `null` if there is no match within `valueSet` for a given key. This boils down to the following possible results for `getValue`:

1. No match for the `value` parameter is found, so the state of `valueSet` is moot; return `null`.
2. There is a match for `value` but `valueSet` is `null`; return matching `value` string.
3. There is a match for `value`, `valueSet` exists, but `getMappedValue` returns null; return `null`.
4. There is a match for `value`, `valueSet` exists, and `getMappedValue` returns a value; return matching `valueSet` string.

Removing the separate null check for `valueSet` would conflate 2 and 3.